packages/core/src/browser/new-element-contribution.ts

```diff
@@ -0,0 +1,153 @@
+        }
+
+        // main menu bar
+        registry.registerSubmenu(NEW_ELEMENT_MAIN_MENU, 'New Element');
+        for (const template of NEW_ELEMENT_TEMPLATES) {
+            registry.registerMenuAction(NEW_ELEMENT_MAIN_MENU, { commandId: template.id, label: template.label + '...' });
+        }
+    }
+
+    protected async createNewElementFile(uri: URI, template: NewElementTemplate): Promise<void> {
+        const parent = await this.getDirectory(uri);
+        if (parent) {
+            const parentUri = parent.resource;
+            const dialog = new WorkspaceInputDialog(
+                {
+                    title: nls.localizeByDefault('New ' + template.label + '...'),
+                    parentUri: parentUri,
+                    initialValue: 'New' + template.label,
+                    placeholder: 'New ' + template.label,
+                    validate: newName => this.validateElementFileName(newName, parent, template)
+                },
+                this.labelProvider
+            );
+            const name = await dialog.open();
+            if (name) {
+                const fileName = this.applyFileExtension(name, template);
+                const elementName = this.removeFileExtension(name, template);
+                const content = typeof template.content === 'string' ? template.content : template.content(elementName);
+                const fileUri = parentUri.resolve(fileName);
+                await this.fileService.create(fileUri, content);
+                this.fireCreateNewFile({ parent: parentUri, uri: fileUri });
+                open(this.openerService, fileUri);
+            }
+        }
+    }
+
+    protected validateElementFileName(name: string, parent: FileStat, template: NewElementTemplate): MaybePromise<DialogError> {
+        // default behavior for empty strings is like cancel
+        if (!name) {
+            return '';
+        }
+        // we automatically may name some part in the initial code after the given name so ensure it is an ID
+        if (!ID_REGEX.test(name)) {
+            return nls.localizeByDefault(`'${name}' is not a valid name, must match: ${ID_REGEX}.`);
+        }
+        // automatically apply file extension for better UX
+        return this.validateFileName(this.applyFileExtension(name, template), parent, true);
+    }
+
+    protected applyFileExtension(name: string, template: NewElementTemplate): string {
+        return name.endsWith(template.fileExtension) ? name : name + template.fileExtension;
+    }
+
+    protected removeFileExtension(name: string, template: NewElementTemplate): string {
+        return name.endsWith(template.fileExtension) ? name.slice(0, -template.fileExtension.length) : name;
+    }
+}
+
+@injectable()
+export class CrossModelFileNavigatorContribution extends FileNavigatorContribution {
+    override registerCommands(registry: CommandRegistry): void {
+        super.registerCommands(registry);
```

Comment:
@martin-fleck-at 
If I understand this correctly this also registers the command for the command pallete?

When I type 'Entity' or 'Relationship' in the command pallete I now see the entries, which seems a bit odd. Also I think the entry should be 'New Entity' for clarity for the user.

Reply:
Good point! I removed the items from the command palette (using an empty label) for now. I think that is a bit cleaner. If you want, however, we can also add them with the 'New X' label.

Reply:
@martin-fleck-at Please add the option in the command palette:
- New Entity
- New Relationship
- New ...

Reply:
Ok, I'll push an update as soon as https://github.com/CrossBreezeNL/crossmodel/pull/28 is merged as I have to update the PR then anyway for the initial file content.